pkg/util/wait/wait_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package wait
+
+import (
+	"errors"
+	"strings"
+	"testing"
+	"time"
+
+	k8swait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestPollSuccess(t *testing.T) {
+	executions := 0
+
+	err := Poll(1*time.Millisecond, 100*time.Millisecond, func() (error, error) {
+		executions++
+		return nil, nil
+	})
+
+	if err != nil {
+		t.Fatalf("Poll should have returned nil, but returned %v", err)
+	}
+
+	if executions > 1 {
```

Comment:
Nit: I'd check that `executions != 1`, i.e. to cover the case when it never runs.